### PR TITLE
editoast: infra/{id}/pathfinding starts in both directions

### LINKED ISSUE #5543: Infinite computing when creating new route
### What happened?

I'm trying to fix a hole in the infrastructure so i created a track link and now i want to add a new route, but looking for compatible routes between start and stop has been computing for 5mn without finding anything
image

### What did you expect to happen?

It should either find a route or return an error.

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to infra editor
2. Select `France_en_qualite`
3. Try to create a new route between this detector
image
And this one
image

### What operating system, browser and environment are you using?

- **Browser**: Chrome
- **OS**: Win 11
- **Env**: staging

### OSRD version (top right corner `Account` button > `Informations`)

1280353d

### PR DESCRIPTION
**Pull request by Tristramg**
Closes #5543

- The first commit searches in both directions and fixes the functional aspects
- The second commit adds a bound to avoid searching routes for too long (this endpoint searches k-shortest path, and the second shortest could be significantly longer)
- The last commit is removing the parameter

### CODE REVIEW

**Review by multun — APPROVED** [review]

File: editoast/src/views/infra/pathfinding.rs [hunk 1]
```diff
@@ -177,14 +184,45 @@ fn compute_path(
...
     let start = &input.starting;
-    let start = PathfindingStep::new_init(start.track.0.clone(), start.position, start.direction);
+    let start = PathfindingStep::new_init(start.track.0.clone(), start.position);
 
     let track_sections = infra_cache.track_sections();
     // Transform a length (in m) into a cost (in mm). This provide the Ord implementation for our cost using u64.
     let into_cost = |length: f64| (length * 100.).round() as u64;
     let get_length = |track: &String| track_sections[track].unwrap_track_section().length;
     let success = |step: &PathfindingStep| step.found;
+    let mut best_distance = u64::MAX;
```
Comment (flomonster) [inline editoast/src/views/infra/pathfinding.rs]: We should set this value to the geographical distance between the points multiplied by two.
Note: We can use the bounding boxes to calculate this.

**Review by flomonster — CHANGES_REQUESTED** [review]
We have a 400 error. 

```
{
  "status": 400,
  "type": "editoast:JsonError",
  "context": {
    "cause": "Json deserialize error: unknown field `direction`, expected `track` or `position` at line 1 column 84"
  },
  "message": "Json deserialize error: unknown field `direction`, expected `track` or `position` at line 1 column 84"
}
```

File: editoast/src/views/infra/pathfinding.rs (hunk 1 above)
Comment (Tristramg) [reply editoast/src/views/infra/pathfinding.rs]: I added a commit doing so.

File: editoast/src/views/infra/pathfinding.rs [hunk 2]
```diff
@@ -177,14 +184,58 @@ fn compute_path(
...
     let success = |step: &PathfindingStep| step.found;
+
+    let starting_track = track_sections[&input.starting.track.0].unwrap_track_section();
+    let ending_track = track_sections[&input.ending.track.0].unwrap_track_section();
+    let best_distance = starting_track
+        .bbox_geo
+        .clone()
+        .union(&ending_track.bbox_geo)
+        .diagonal_length();
+    // We build an upper bound that is the diagonal of the bounding bex covering start and end
```
Comment (bougue-pe) [inline editoast/src/views/infra/pathfinding.rs]:
```suggestion
    // We build an upper bound that is the diagonal of the bounding box covering start and end
```

Comment (Tristramg) [reply editoast/src/views/infra/pathfinding.rs]: Thank you!

**Review by SharglutDev — APPROVED** [review]
Lgtm for the front part :)

**Review by SharglutDev — CHANGES_REQUESTED** [review]
Actually front looks good but got the same error as @flomonster  when trying to search for routes

**Review by SharglutDev — APPROVED** [review]
Ok my bad, after rebuilding editoast, its seems ok :) 

image

**Review by flomonster — CHANGES_REQUESTED** [review]
I still have a problem, some valid existing routes can't be recomputed.
I've checked that this isn't due to the geographical max distance.

**Review by flomonster — APPROVED** [review]
Lgtm

**Review by Math-R — APPROVED** [review]

**Pull request merged**